packages/sdk/src/scopes/components/full-screen/full-screen.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  getStorageValue,
+  setStorageValue,
+  type FullScreenErrorStatus,
+} from '@telegram-apps/bridge';
+import { isPageReload } from '@telegram-apps/navigation';
+import { signal } from '@telegram-apps/signals';
+
+import { request } from '@/scopes/globals.js';
+import { createIsSupported } from '@/scopes/toolkit/createIsSupported.js';
+import { createWrapComplete } from '@/scopes/toolkit/createWrapComplete.js';
+import { createWrapSupported } from '@/scopes/toolkit/createWrapSupported.js';
+
+import { FullScreenStatus } from "@/scopes/components/full-screen/types.js";
+
+type StorageValue = boolean;
+
+const REQUEST_METHOD_NAME = 'web_app_request_fullscreen';
+const EXIT_METHOD_NAME = 'web_app_exit_fullscreen';
+
+const CHANGED_EVENT_NAME = 'fullscreen_changed';
+const FAILED_EVENT_NAME = 'fullscreen_failed';
+
+const COMPONENT_NAME = 'fullScreen';
+
+/**
+ * Signal indicating if the Full Screen component is currently mounted.
+ */
+export const isMounted = signal(false);
+
+/**
+ * Signal indicating if the Full Screen is supported by telegram version.
+ */
+export const isSupported = createIsSupported(REQUEST_METHOD_NAME) || createIsSupported(EXIT_METHOD_NAME);
+
+/**
+ * Signal indicating if fullscreen mode is active.
+ */
+export const isFullScreen = signal(false);
+
+const wrapSupported = createWrapSupported(COMPONENT_NAME, REQUEST_METHOD_NAME);
+const wrapComplete = createWrapComplete(COMPONENT_NAME, isMounted, REQUEST_METHOD_NAME);
+
+/**
+ * Requests full screen mode.
+ * @since Mini Apps v8.0
+ * @throws {TypedError} ERR_UNKNOWN_ENV
+ * @throws {TypedError} ERR_NOT_INITIALIZED
+ * @throws {TypedError} ERR_NOT_SUPPORTED
+ * @throws {TypedError} ERR_NOT_MOUNTED
+ * @returns {Promise<FullScreenStatus>} current status of full screen, boolean or error
+ * @example
+ * if (requestFullScreen.isAvailable()) {
+ *   requestFullScreen();
+ * }
+ */
+export const requestFullScreen = wrapComplete('requestFullScreen', (): Promise<FullScreenStatus> => {
+  return setFullScreenMode(true);
+});
```

Comment:
ALREADY_FULLSCREEN result is here just to match TG APIs
By the way, lets make **telegram-apps** a better version of **telegram sdks** 🙂

p.s. for the UNSUPPORTED result, makes sense to throw instead of returning 'result'. 
For the state itself, I couldn't reproduce it, maybe on some devices telegram reports fullscreen unsupported.